src/storage/disk_table.cc

```diff
@@ -1097,8 +1097,128 @@ void DiskTableRowIterator::SeekToFirst() {
+        }
+
+        if (table_index_.AddIndex(index_def) < 0) {
+            PDLOG(WARNING, "add index failed. tid %u pid %u", id_, pid_);
+            return false;
+        }
+
+        index_def->SetInnerPos(inner_id);
+        std::vector<std::shared_ptr<IndexDef>> index_vec = {index_def};
+        auto inner_index_st = std::make_shared<InnerIndexSt>(inner_id, index_vec);
+        table_index_.AddInnerIndex(inner_index_st);
+        table_index_.SetInnerIndexPos(new_table_meta->column_key_size() - 1, inner_id);
+
+        rocksdb::ColumnFamilyOptions cfo;
+
+        if (storage_mode_ == ::openmldb::common::StorageMode::kSSD) {
+            cfo = rocksdb::ColumnFamilyOptions(ssd_option_template);
+        } else {
+            cfo = rocksdb::ColumnFamilyOptions(hdd_option_template);
+        }
+
+        cfo.comparator = &cmp_;
+        cfo.prefix_extractor.reset(new KeyTsPrefixTransform());
+
+        if (index_def->GetTTLType() == ::openmldb::storage::TTLType::kAbsoluteTime ||
+            index_def->GetTTLType() == ::openmldb::storage::TTLType::kAbsOrLat) {
+            cfo.compaction_filter_factory = std::make_shared<AbsoluteTTLFilterFactory>(inner_index_st);
+        }
+
+        rocksdb::ColumnFamilyHandle* handle;
+        cf_ds_.push_back(rocksdb::ColumnFamilyDescriptor(index_def->GetName(), cfo));
+        DEBUGLOG("add cf_name %s. tid %u pid %u", index_def->GetName().c_str(), id_, pid_);
+
+        rocksdb::Status s = db_->CreateColumnFamily(cfo, index_def->GetName(), &handle);
+        if (!s.ok()) {
+            PDLOG(WARNING, "rocksdb failed to create ColumnFamily. tid %u pid %u error %s",
+                  id_, pid_, s.ToString().c_str());
+            return false;
+        }
+
+        cf_hs_.push_back(handle);
+    }
+    index_def->SetStatus(IndexStatus::kReady);
+    std::atomic_store_explicit(&table_meta_, new_table_meta, std::memory_order_release);
+    return true;
+}
+
 bool DiskTable::DeleteIndex(const std::string& idx_name) {
-    // TODO(litongxin)
+    std::shared_ptr<IndexDef> index_def = table_index_.GetIndex(idx_name);
+    if (!index_def) {
+        PDLOG(WARNING, "index %s is not exist. tid %u pid %u", idx_name.c_str(), id_, pid_);
+        return false;
+    }
+    if (index_def->GetId() == 0) {
+        PDLOG(WARNING, "index %s is primary key, cannot delete. tid %u pid %u", idx_name.c_str(), id_, pid_);
+        return false;
+    }
+    if (!index_def->IsReady()) {
+        PDLOG(WARNING, "index %s can't delete. tid %u pid %u", idx_name.c_str(), id_, pid_);
+        return false;
+    }
+    auto table_meta = GetTableMeta();
+    auto new_table_meta = std::make_shared<::openmldb::api::TableMeta>(*table_meta);
+    if (index_def->GetId() < (uint32_t)table_meta->column_key_size()) {
+        new_table_meta->mutable_column_key(index_def->GetId())->set_flag(1);
+    }
+    std::atomic_store_explicit(&table_meta_, new_table_meta, std::memory_order_release);
+    index_def->SetStatus(IndexStatus::kWaiting);
```

Comment:
The data in these index should be delete

Reply:
@jiang1997 in mem table, it is deleted in `SchedGc`